<launch>
  <arg name="use_sim" default="true" />

  <node name="box_detector_node" pkg="robot_manipulation" type="box_detector" output="screen">
    <remap from="/point_cloud" to="/camera/depth/color/points" />
    <remap from="/rviz_debug" to="/box_detector/rviz_debug" />
    <remap from="/filtered_point_cloud" to="/box_detector/filtered" />
    <remap from="/detections" to="/box_detector/detections" />
  </node>

  <node name="object_reg_node" pkg="robot_manipulation" type="object_find2" output="screen">
    <remap from="/point_cloud" to="/filter_cloud/filtered" />
    <remap from="/rviz_debug" to="/object_reg/rviz_debug" />
    <remap from="/filtered_point_cloud" to="/object_reg/filtered" />
    <remap from="/detections" to="/object_reg/detections" />
  </node>

  <node name="filter_cloud_node" pkg="robot_manipulation" type="filter_cloud" output="screen">
    <remap from="/point_cloud_1" to="/camera/depth/color/points" />
    <remap from="/point_cloud_2" to="/camera/depth/color/points" />
    <remap from="/filtered_point_cloud" to="/filter_cloud/filtered" />
    <!-- Parameters for cropping -->
    <param name="crop_min_x" type="double" value="0.0" />
    <param name="crop_min_y" type="double" value="-0.17" />
    <param name="crop_min_z" type="double" value="0.0" />
    <param name="crop_max_x" type="double" value="0.5" />
    <param name="crop_max_y" type="double" value="0.19" />
    <param name="crop_max_z" type="double" value="0.15" />
    <!-- Parameter for downsampling -->
    <param name="voxel_size" type="double" value="0.003" />
    <!-- Parameter for target frame -->
    <param name="base_link_frame" type="str" value="wx250s/base_link" />
  </node>

  <node name="packer" pkg="robot_manipulation" type="packer2.py" output="screen"></node>

  <!-- Launch the base functionality -->
  <include file="$(find robot_manipulation)/launch/base_old.launch">
    <arg name="use_rviz" value="true" />
    <arg name="use_sim" value="$(arg use_sim)" />
    <arg name="dual_camera" value="true" />
  </include>
</launch>
